File: Interview Testing/fib.py
#recursive method to find the nth number of the Fibonacci sequence
#Potential interview question
def fibonacci(n):
    if n < 0:
        print("You have entered an invalid integer. Please choose a positive integer.")
        return -1
    if n <= 1:
        return n # NOTE: should only trigger for n=1 or n=0
    else:
        return fibonacci(n-1) + fibonacci(n-2)

#Method that builds off of fibonacci to find the first n numbers in the sequence
def fibonacci_list(n):
    first_n_nums = []
    for i in range(n):
        first_n_nums.append(fibonacci(i))
    return first_n_nums

#main function for testing
if __name__ == "__main__":
    n = int(input("Please choose a positive integer. "))
    print("The nth number is " + str(fibonacci(n)))
    print(fibonacci_list(n))

File: Interview Testing/fib.java
class fib_sequence{
  static int fib(int n){
    if (n < 0){
      System.out.println("You have entered an invalid number");
      return -1;
    }
    if(n <= 1){
      return n;
    }
    else{
      return fib(n-1) + fib(n-2);
    }
  }
  static void fibonacci_list(int n){
    for (int i = 0; i < n; i++){
      System.out.println(fib(i));
    }
  }
  public static void main (String args[]){
    int n = 9;
    System.out.println(fib(n));
    fibonacci_list(n);
  }
}
